src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,9 +1,270 @@
+  };
+
+  hideBonus = () => {
+    const { isBonus } = this.state;
+    this.setState({ isBonus: !isBonus });
+  };
+
+  hideShoppingBag = () => {
+    const { isShoppingBag } = this.state;
+    this.setState({ isShoppingBag: !isShoppingBag });
+  };
+
+  changePhoto = photo => {
+    this.setState({
+      imageNum: photo,
+    });
+  };
+
+  gotoCart = () => {
+    this.props.history.push('/cart');
+  };
+
+  gotoMain = () => {
+    this.props.history.push('/mainPage');
+  };
+
   render() {
-    return <div></div>;
+    const { productInfoList, size, isSize, isBonus, isShoppingBag, imageNum } =
+      this.state;
+    return (
+      <div className="productDetail">
+        <div className="nav">임시</div>
+        <div className="list">
+          <span className="home" onClick={this.gotoMain}>
+            홈
+          </span>
+          &nbsp;&nbsp;/&nbsp;&nbsp;
+          <span className="category">
+            {productInfoList && productInfoList.section}
+          </span>
+          &nbsp; / &nbsp;
+          <span className="category">
+            {productInfoList && productInfoList.category}
+          </span>
+          &nbsp; / &nbsp;
+          {productInfoList && productInfoList.name}
+        </div>
+        <div className="productMain">
+          <div className="productMainInteraction">
+            <div className="slideBox">
+              <div className="miniPhotoBox">
+                {productInfoList.images &&
+                  productInfoList.images.map((image, idx) => {
+                    return (
+                      <div className="miniPhoto">
+                        <ProductPhoto
+                          id={image.id}
+                          imageUrl={image.image_url}
+                          imageNum={-684 * idx}
+                          changePhoto={this.changePhoto}
+                          key={productInfoList.images.id}
+                        />
+                      </div>
+                    );
+                  })}
+              </div>
+              <div
+                className="slideList"
+                style={{
+                  transform: `translate3d(
+                ${imageNum}px, 0px, 0px`,
+                }}
+              >
+                {productInfoList.images &&
+                  productInfoList.images.map(image => {
+                    return (
+                      <span className="slideContent" key="image.id">
+                        <ProductPhoto imageUrl={image.image_url} />
+                      </span>
+                    );
+                  })}
+              </div>
+            </div>
+            <div className="productText">
+              <div className="productName">
+                {productInfoList && productInfoList.name}
+              </div>
+              <div className="productPrice">
+                <div className="productColor">
+                  {productInfoList.colors && productInfoList.colors[0].name}
```

Comment:
```suggestion
                  {productInfoList.colors?.[0].name}
```
이렇게 사용하면 동일한 의미가 되겠네요

Reply:
넵 수정했습니다 🔥